src/main/java/org/testng/internal/Parameters.java

```diff
@@ -62,6 +62,7 @@
 
   private static final Map<Class<? extends Annotation>, Class<? extends IAnnotation>> ANNOTATION_MAP =
           new ConcurrentHashMap<>();
+  private static boolean isInjected = false;
```

Comment:
would this be a class level static data member or should this be a method's local variable ?

Reply:
Yeah `static` looks wrong here.

Reply:
You are right. I will remove this filed and use 
`if (optionalValues.length != parameterNames.length)`
in `Parameters::createParam` to judge the condition.